drivers/can/can_mcan.c

```diff
@@ -292,7 +292,7 @@ int can_mcan_start(const struct device *dev)
 {
 	const struct can_mcan_config *config = dev->config;
 	struct can_mcan_data *data = dev->data;
-	int err;
+	int err = 0;
```

Comment:
Just for my understanding.

Is this change really required? I don't see any code path where err would be read before getting assigned to some value.

Reply:
I was thinking the same. But I thought it would be OK to keep it for clarity reasons. We are returning `err` at the end now, so one one does not have to check every path in the code whether `err` is guaranteed to have a valid value.